nixos/modules/services/misc/overseerr.nix

```diff
@@ -0,0 +1,64 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+let
+  cfg = config.services.overseerr;
+in
+{
+  meta.maintainers = [ maintainers.caarlos0 ];
+
+  options.services.overseerr = {
+    enable = mkEnableOption (mdDoc ''
+      Request management and media discovery tool for the Plex ecosystem
+    '');
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = mdDoc ''Open port in the firewall for the Overseerr web interface.'';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 5055;
+      description = mdDoc ''The port which the Overseerr web UI should listen to.'';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.overseerr = {
+      description = "Request management and media discovery tool for the Plex ecosystem";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      environment.PORT = toString cfg.port;
+      serviceConfig = {
+        Type = "exec";
+        StateDirectory = "overseerr";
+        WorkingDirectory = "${pkgs.overseerr}/libexec/overseerr/deps/overseerr";
+        DynamicUser = true;
+        ExecStart = "${pkgs.overseerr}/bin/overseerr";
+        BindPaths = [ "/var/lib/overseerr/:${pkgs.overseerr}/libexec/overseerr/deps/overseerr/config/" ];
```

Comment:
What's going on here? Does it expect read and write config in `libexec`?

In that case, I think it would warrant patching the package. Packages are supposed to be usable outside of nixos as well, and I wouldn't really consider packages that requires bindmounts into the nix store to be in a working condition.

Reply:
It's been a while and I don't remember it that well, but I think it does require writing to that dir, yes.

It seems like we can set a `CONFIG_DIRECTORY` environment variable... what's the recommended thing to do here? Ask the user where to put it?